src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -1403,13 +1403,27 @@ private bool HadAmbiguousWorseMethods(BindingDiagnosticBag diagnostics, Immutabl
             }
             else
             {
-                // error CS0121: The call is ambiguous between the following methods or properties: 'P.W(A)' and 'P.W(B)'
-                diagnostics.Add(
-                    CreateAmbiguousCallDiagnosticInfo(
-                        worseResult1.LeastOverriddenMember.OriginalDefinition,
-                        worseResult2.LeastOverriddenMember.OriginalDefinition,
-                        symbols),
-                    location);
+                if (ReferenceEquals(worseResult1.LeastOverriddenMember.OriginalDefinition, worseResult2.LeastOverriddenMember.OriginalDefinition) &&
+                    worseResult1.LeastOverriddenMember.ContainingType.IsExtension) // PROTOTYPE consider using the improved diagnostic for scenarios beyond extensions
+                {
+                    // error CS0121: The call is ambiguous between the following methods or properties: 'P.W(A)' and 'P.W(B)'
+                    diagnostics.Add(
+                        CreateAmbiguousCallDiagnosticInfo(
+                            worseResult1.LeastOverriddenMember,
```

Comment:
>LeastOverriddenMember [](http://example.com/codeflow?start=41&length=21)

I am not sure what was the rational in the existing code to use ```OriginalDefinition```. Perhaps there was a desire to exclude method type arguments, ```ConstructedFrom``` should be used then. Since you are not changing existing logic, we losing the benefit of observing effect of the change on the broader set of scenarios. In order to make sure that we are not missing something obvious, consider adding a test with an ambiguity between generic methods.